test/cmd-collect-http.test.js

```diff
@@ -0,0 +1,31 @@
+'use strict'
+
+const test = require('tap').test
+const http = require('http')
+const CollectAndRead = require('./collect-and-read.js')
+
+test('cmd - collect - detect server port', function (t) {
+  const cmd = new CollectAndRead({http: true}, '-e', `
+    const http = require('http')
+    http.createServer(onrequest).listen(0)
+
+    function onrequest (req, res) {
+      this.close()
+      res.end('from server')
+    }
+  `)
+
+  cmd.tool.on('listening', function (port) {
+    t.ok(typeof port === 'number')
+    t.ok(port > 0)
+
+    http.get(`http://127.0.0.1:${port}`, function (res) {
```

Comment:
how can a user specify the path of the http request?

Reply:
would happen in clinic itself `--http {optionally-set-path-here}`.

this simply exposes a way to for a module consumer to know what port a server has bound to

Reply:
perfect.